gigantum/gigantum-client
# On token verify error, refetch auth config and better ssl error message

## Pull Request for Issue #1770

**Describe changes introduced by this PR (Bugs fixed, features added, docs updated, etc.)**:
This PR makes 2 small changes to the auth workflow. First, on token verification errors, the auth well-known data is re-fetched. If the fetch succeeds the cache and persisted file are updated. Since the jwks is also removed, it'll refetch everything automatically when the user tries to log in again. Second, a better error message was added when there is an ssl error during the auth process (almost always when fetching the jwks). This is a super common error mode for enterprise users who have a team server that requires some custom CAs 


**QA Test Steps and Exit Criteria**:
- Try logging into a team server that requires a CA to be configured for the first time and you should get the SSL warning. One of the test servers supports this (ask @dkleissa if you need help)
- Modify the persisted server file for a team server you are logged into, then corrupt your token. WHen you try to do something with the API you should see your identity get removed and the server file should update to be correct.
